oneflow/api/cpp/framework/graph.cpp

```diff
@@ -0,0 +1,353 @@
+  }
+  auto* ctx = JUST(of::GetCurInferCtx());
+  JUST(ctx->AddAndInferConsistentOp(op_conf));
+  return of::Maybe<void>::Ok();
+}
+
+of::Maybe<void> Graph::GraphImpl::BuildGraph(const std::vector<Tensor>& inputs) {
+  CompileScope build_graph_scope(job_.job_conf(), *device_.device_->shared_from_symbol(),
+                                 xrt_kind_);
+  {
+    // TODO(zzk0): remove this; used for input tensor order
+    int input_tensor_order = 0;
+    const of::OpGraph op_graph(job_);
+    JUST(op_graph.ForEachOpNode([&](const of::OpNode& node) -> of::Maybe<void> {
+      const of::OperatorConf& op_conf = node.op().op_conf();
+      JUST(AddOp(op_conf));
+      if (op_conf.has_input_conf()) {
+        // TODO(zzk0): input tensor order
+        input_name_to_tensor_[op_conf.name()] = inputs.at(input_tensor_order++).tensor_;
+      } else if (op_conf.has_variable_conf()) {
+        const of::LazyMode::Guard lazy_mode_disabled_guard{false};
+        const of::VariableOpConf variable_conf = op_conf.variable_conf();
+        variable_op_name_to_tensor_[op_conf.name()] = JUST(of::one::functional::Empty(
+            of::Shape(variable_conf.shape()),
+            JUST(of::DType::Get(static_cast<of::DataType>(variable_conf.data_type()))),
+            *device_.device_));
+      }
+      return of::Maybe<void>::Ok();
+    }));
+  }
+  JUST(of::CurJobBuildAndInferCtx_Complete());
+  {
+    const std::shared_ptr<of::Job> complete_job = JUST(of::GetCurrentJob());
+    const of::OpGraph complete_graph(*complete_job);
+    JUST(complete_graph.ForEachOpNode([&](const of::OpNode& node) -> of::Maybe<void> {
+      const of::LazyMode::Guard lazy_mode_disabled_guard{false};
+      const of::OperatorConf& op_conf = node.op().op_conf();
+      if (op_conf.has_output_conf()) {
+        of::InterfaceBlobConf blob_conf = op_conf.output_conf().blob_conf();
+        if (batch_size_ > 0) {
+          const std::string input_lbi_str = op_conf.output_conf().in();
+          const of::LogicalBlobId input_lbi = of::GenLogicalBlobId(input_lbi_str);
+          int64_t batch_size = node.LogicalBlobDesc4Lbi(input_lbi).shape().At(0);
+          blob_conf.mutable_shape()->set_dim(0, batch_size);
+        }
+        output_name_to_tensor_[op_conf.name()] = JUST(of::one::functional::Empty(
+            of::Shape(blob_conf.shape()),
+            JUST(of::DType::Get(static_cast<of::DataType>(blob_conf.data_type()))),
+            *device_.device_));
+      }
+      return of::Maybe<void>::Ok();
+    }));
+  }
+  return of::Maybe<void>::Ok();
+}
+
+of::Maybe<void> Graph::GraphImpl::LoadCheckpoint() {
+  for (const auto& variable_op_name_and_tensor : variable_op_name_to_tensor_) {
+    const auto& variable_op_name = variable_op_name_and_tensor.first;
+    const auto& variable_tensor = variable_op_name_and_tensor.second;
+    const std::string variable_filename = model_path_ + "/" + variable_op_name + "/out";
+    const std::string buffer = [&variable_filename]() {
```

Comment:
这里顺便这样改一下吧，当时我写的不够好，这里这个 lambda 的角色就是应该用 & 来捕获变量

```suggestion
    const std::string buffer = [&]() {
```